python/pyarrow/dataset.py

```diff
@@ -780,5 +784,5 @@ def write_dataset(data, base_dir, basename_template=None, format=None,
 
     _filesystemdataset_write(
         scanner, base_dir, basename_template, filesystem, partitioning,
-        file_options, max_partitions
+        file_options, use_threads, use_async, max_partitions
```

Comment:
If I'm not mistaken this'll be an error since those parameters don't exist?

Reply:
They should get passed into L744/L777 above, most likely.

Reply:
Just fixed.

Reply:
Hmm, `use_threads` wasn't being passed in on 744.  I don't think there's any need to pass either since it's just scanning in memory.  I'm not entirely certain on `use_threads` though.